manifests/init.pp

```diff
@@ -7,69 +7,123 @@
-  Optional[String[1]]          $zpool                 = undef,
-  Stdlib::Absolutepath         $zrootfs               = '/poudriere',
-  String[1]                    $freebsd_host          = 'http://ftp.freebsd.org/',
-  Stdlib::Absolutepath         $resolv_conf           = '/etc/resolv.conf',
-  Boolean                      $ccache_enable         = false,
-  Stdlib::Absolutepath         $ccache_dir            = '/var/cache/ccache',
-  Stdlib::Absolutepath         $poudriere_base        = '/usr/local/poudriere',
-  String[1]                    $poudriere_data        = '${BASEFS}/data',
-  Enum['yes', 'no']            $use_portlint          = 'no',
-  Optional[String[1]]          $mfssize               = undef,
-  Enum['yes', 'no']            $tmpfs                 = 'yes',
-  Stdlib::Absolutepath         $distfiles_cache       = '/usr/ports/distfiles',
-  Optional[String[1]]          $csup_host             = undef,
-  Optional[String[1]]          $svn_host              = undef,
-  Enum['yes', 'no', 'verbose'] $check_changed_options = 'verbose',
-  Enum['yes', 'no']            $check_changed_deps    = 'yes',
-  Optional[String[1]]          $pkg_repo_signing_key  = undef,
-  Integer[1]                   $parallel_jobs         = $facts['processors']['count'],
-  Optional[String[1]]          $save_wrkdir           = undef,
-  Optional[String[1]]          $wrkdir_archive_format = undef,
-  Optional[String[1]]          $nolinux               = undef,
-  Optional[String[1]]          $no_package_building   = undef,
-  Optional[String[1]]          $no_restricted         = undef,
-  Optional[String[1]]          $allow_make_jobs       = undef,
-  Optional[String[1]]          $url_base              = undef,
-  Optional[String[1]]          $max_execution_time    = undef,
-  Optional[String[1]]          $nohang_time           = undef,
-  Optional[String[1]]          $http_proxy            = undef,
-  Optional[String[1]]          $ftp_proxy             = undef,
-  Optional[String[1]]          $build_as_non_root     = undef,
-  Hash                         $environments          = {},
-  Hash                         $portstrees            = {},
-  String[1]                    $xbuild_package        = 'qemu-user-static',
+  Optional[String[1]]                $zpool                      = undef,
+  Stdlib::Absolutepath               $zrootfs                    = '/poudriere',
+  String[1]                          $freebsd_host               = 'http://ftp.freebsd.org/',
+  Stdlib::Absolutepath               $resolv_conf                = '/etc/resolv.conf',
+  Stdlib::Absolutepath               $poudriere_base             = '/usr/local/poudriere',
+  String[1]                          $poudriere_data             = '${BASEFS}/data',
+  Enum['yes', 'no']                  $use_portlint               = 'no',
+  Optional[String[1]]                $mfssize                    = undef,
+  String[1]                          $tmpfs                      = 'yes',
```

Comment:
which values are supported here besides `yes`?

Reply:
```
# Use tmpfs(5)
# This can be a space-separated list of options:
# wrkdir    - Use tmpfs(5) for port building WRKDIRPREFIX
# data      - Use tmpfs(5) for poudriere cache/temp build data
# localbase - Use tmpfs(5) for LOCALBASE (installing ports for packaging/testing)
# all       - Run the entire build in memory, including builder jails.
# yes       - Enables tmpfs(5) for wrkdir and data
# no        - Disable use of tmpfs(5)
# EXAMPLE: USE_TMPFS="wrkdir data"
```
I guess this list does not move much… Not sure if it is worth the effort to change this to some kind of `Variant[Enum['all', 'no', 'yes'],Array[Enum['data', 'localbase', 'wrkdir']]]` and adjust the code accordingly.  What do you think?

Reply:
can you please create a custom type for this and use it?

Reply:
Here we go! Added as a86e105fae3ff691fee1f7b7bbbc1ae4e739667b